onnxruntime/core/providers/rocm/miopen_common.h

```diff
@@ -14,6 +14,14 @@ const double MIOPEN_BN_MIN_EPSILON = 1e-5;
 namespace onnxruntime {
 namespace rocm {
 
+// Define miopenTensorLayout_t there if ROCm version < 5.3.0.
+#ifndef USE_MIOPEN_TENSOR_LAYOUT
+typedef enum {
+  miopenTensorNCHW = 0,
+  miopenTensorNHWC = 1,
+} miopenTensorLayout_t;
```

Comment:
Isn't this ultimately cause compilation error, because it will be defined in miopen's header after some version.

Reply:
Yes, it will cause compilation error if you install new version of MIOpen(> 5.2.3) with ROCm version( < 5.2.3).

Reply:
How about `#ifdef MIOPEN_VERSION < blahblah`?

Reply:
Done.